features/openconfig/telemetry-client/src/main/java/org/opennms/features/openconfig/telemetry/OpenConfigClientImpl.java

```diff
@@ -78,16 +87,24 @@
     private ManagedChannel channel;
     private final InetAddress host;
     private Integer port;
-    private Map<String, String> parameters = new HashMap<>();
+    private String mode;
+    private Integer interval;
+    private Integer retries;
+    private List<Map<String,String>> paramList = new ArrayList<>();
     private AtomicBoolean closed = new AtomicBoolean(false);
     private AtomicBoolean scheduled = new AtomicBoolean(false);
     private ExecutorService executor = Executors.newSingleThreadExecutor();
     private ScheduledExecutorService scheduledExecutor = Executors.newSingleThreadScheduledExecutor();
 
-    public OpenConfigClientImpl(InetAddress host, Map<String, String> params) {
+    public OpenConfigClientImpl(InetAddress host, List<Map<String, String>> paramList) {
         this.host = Objects.requireNonNull(host);
-        this.port = Objects.requireNonNull(StringUtils.parseInt(params.get("port"), null));
-        parameters.putAll(params);
+        this.paramList.addAll(paramList);
+        // Extract port and mode which are global.
+        this.paramList.stream().filter(entry -> entry.get("port") != null)
```

Comment:
I think it would be better to enforce en empty group on the global parameters. (And maybe print an error log message if they are specified with a group)

Reply:
At this point, Client doesn't have any group info. It has a set of parameters.